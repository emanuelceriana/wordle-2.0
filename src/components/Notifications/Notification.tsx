import { ReactNode } from "react";
import styles from "./Notification.module.scss";
import cn from "classnames";

interface NotificationProps {
  message: ReactNode;
}

export const Notification = ({ message }: NotificationProps) => {
  return (
    message && (
      <div className={styles.notification}>
        <div
          data-testid="notificationMessage"
          className={cn(styles.message, styles.animation)}
        >
          <p>{message}</p>
        </div>
      </div>
    )
  );
};
