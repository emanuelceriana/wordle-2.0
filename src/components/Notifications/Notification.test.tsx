import { render } from "@testing-library/react";
import { Notification } from "./Notification";

describe("<Notification />", () => {
  const testMessage = "test";

  it("should render without errors", () => {
    const { asFragment } = render(<Notification message={testMessage} />, {});
    expect(asFragment()).toMatchSnapshot();
  });
  it("should render the message correctly", () => {
    const { getByText } = render(<Notification message={testMessage} />, {});
    expect(getByText(testMessage)).toBeInTheDocument();
  });
});
