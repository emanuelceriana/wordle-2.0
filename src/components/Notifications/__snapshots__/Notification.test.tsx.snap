// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> should render without errors 1`] = `
<DocumentFragment>
  <div
    class="notification"
  >
    <div
      class="message animation"
      data-testid="notificationMessage"
    >
      <p>
        test
      </p>
    </div>
  </div>
</DocumentFragment>
`;
